src/MyMarketManager.Data.Processing/StagingBatchHandler.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using MyMarketManager.Data;
+using MyMarketManager.Data.Enums;
+
+namespace MyMarketManager.Data.Processing;
+
+/// <summary>
+/// Handler that fetches and processes queued staging batches.
+/// This replaces the old BatchProcessingService logic.
+/// </summary>
+public class StagingBatchHandler : IWorkItemHandler<StagingBatchWorkItem>
+{
+    private readonly MyMarketManagerDbContext _context;
+    private readonly IBatchProcessorFactory _factory;
+    private readonly ILogger<StagingBatchHandler> _logger;
+
+    public StagingBatchHandler(
+        MyMarketManagerDbContext context,
+        IBatchProcessorFactory factory,
```

Comment:
Can we maybe also merge this concept so instead of registering batch processors and the work item handler, we can register multiple handlers. This way we can rather have a single source.

Instead of getting items from the db and finding the processor, the processor only gets the items that are handled by that specific processor.

So instead of a staging batch processor, we will have a shein batch processor and a yoco batch processor.

The registration can then indicate that this one is an ingestion or internal or export processor. Or any other metadata.

The UI can then query for all ingestion or all export. The registered name can also be used to load any configuration options.